lock.go

```diff
@@ -63,8 +63,6 @@ func readLock(r io.Reader) (*lock, error) {
 			v = gps.NewBranch(ld.Branch).Is(r)
 		} else if r == "" {
 			return nil, fmt.Errorf("lock file has entry for %s, but specifies no version", ld.Name)
-		} else {
-			v = r
```

Comment:
I think the lint warning is correct. Dropping the else and having `v = r` would be the exact same.

Regarding the style, I used to think like you. Then, premature optimization :) After reading stuff like this on the std itself often, I came to realise it's better to write the simple version and expect the compiler (e.g. SSA) to actually avoid the extra work.

```
a := defaultVal
if someCond {
    a = otherVal
}
```